src/theory/arith/nl/ext/arith_nl_compare_proof_gen.cpp

```diff
@@ -225,13 +261,12 @@ std::shared_ptr<ProofNode> ArithNlCompareProofGenerator::getProofFor(Node fact)
       expcFinal.emplace_back(expc[i]);
     }
   }
-  Node opa = mkProduct(nm, cprodt[0]);
-  Node opb = mkProduct(nm, cprodt[1]);
-  Node newConc = mkLit(nm, ck, opa, opb);
-  Trace("arith-nl-compare")
-      << "...processed prove: " << expc << " => " << concc << std::endl;
+  ProofChecker* pc = d_env.getProofNodeManager()->getChecker();
+  Node newConc =
+      pc->checkDebug(ProofRule::ARITH_MULT_ABS_COMPARISON, expcFinal, {});
```

Comment:
Is this necessary?

Reply:
It simplifies the code quite a bit. Instead of trying to compute what the target conclusion should be, we reference the proof checker to set up the `MACRO_SR_PRED_TRANSFORM` step below.